pulpcore/cli/ansible/remote.py

```diff
@@ -0,0 +1,232 @@
+import click
+
+from typing import Any, Dict, Optional, cast, Callable, TypeVar
+
+from pulpcore.cli.common.generic import (
+    list_entities,
+    show_by_name,
+    destroy_by_name,
+)
+from pulpcore.cli.common.context import (
+    PulpContext,
+    pass_pulp_context,
+    pass_entity_context,
+    PulpEntityContext,
+)
+from pulpcore.cli.ansible.context import (
+    PulpAnsibleRoleRemoteContext,
+    PulpAnsibleCollectionRemoteContext,
+)
+
+RT = TypeVar("RT")  # return type
+
+
+def _type_callback(ctx: click.Context, param: Any, value: str) -> str:
```

Comment:
@mdellweg This works pretty well for the two remote types. The one thing is that the --help option will show the options from both commands despite what type is selected. Not sure if it's a bug or intended behaviour.

Reply:
Please don't invest too much into this right now. Just keep it working for now. I am experimenting with a more general approach here. And it seems to do just fine according to the help screen.

https://github.com/pulp/pulp-cli/pull/64/files#diff-2330f1d2bec4c88428af0e90f4ad6feff6a1663af7be7b9fedb645ed98846f68R18-R42